### PR TITLE
Add donation form validation of Custom amounts (liberapay#1696)

### LINKED ISSUE #1696: Warn when custom amount is invalid
Steps to replicate
------
1. Click "Donate" on any project's page
2. Choose "Custom"
3. Don't input any value in the "amount" field or input an invalid value (e.g. "abc").
4. Click "Donate" 

Current behavior
-----
The page "Bad Request" appears, saying '"" is not a number'.

Desired behavior
-----
After clicking "Donate", the "Amount" field should be highlighted in red, with an error message shown right below the field.

### PR DESCRIPTION
**Pull request by 1124816**
Closes #1696. 
This is my first contribution so pointers very welcome. I wasn't sure how best to make the error message translatable so I made up an html attribute.

### CODE REVIEW

**Comment by 1124816** [comment]
I think this is what you are asking for? Should I squash the commits?

File: js/10-base.js [hunk 1]
```diff
@@ -93,6 +93,17 @@ Liberapay.init = function() {
         ).prop('selected', true);
         history.pushState(null, null, location.pathname + href + location.hash);
     });
+    
+    $('input[data-required-if-checked]').each(function(index, value) {
+        var radio = $(value).attr('data-required-if-checked');
+        $($(radio).parents('form')).on('click', function(e) {
```
Comment (Changaco) [inline js/10-base.js:99]: Clicking isn't the only way to interact with a form. The proper solution is probably to listen for the [`change`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) or [`input`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event) event on the radio input(s).

File: js/10-base.js [hunk 2]
```diff
@@ -93,6 +93,17 @@ Liberapay.init = function() {
...
+            if($(radio).is(':checked')) {
+                $(value).prop('required', true);
+            } else {
+                $(value).prop('required', false);
+            }
```
Comment (Changaco) [inline js/10-base.js:104]: This `if … else …` block can be simplified to `$(value).prop('required', $(radio).is(':checked'));`.

**Comment by 1124816** [comment]
Thanks!

**Pull request merged**